MultiSource/Benchmarks/Olden/mst/hash.h

```diff
@@ -1,20 +1,26 @@
 /* For copyright information, see olden_v1.0/COPYRIGHT */
 
+#include <stdchecked.h>
 #include "stdio.h"
+#include <stdio_checked.h>
 
-typedef struct hash_entry {
+struct hash_entry {
   unsigned int key;
   void *entry;
-  struct hash_entry *next;
-} *HashEntry;
+  ptr<struct hash_entry> next;
+};
 
-typedef struct hash {
-  HashEntry *array;
-  int (*mapfunc)(unsigned int);
+typedef ptr<struct hash_entry> HashEntry;
+
+struct hash {
+  HashEntry *array : itype(array_ptr<HashEntry>);
```

Comment:
The correct type to give it is `array_ptr<HashEntry> array : count(size);` but this currently fails during code generation, as we discussed this morning (the inferred bounds expression contains a declrefexpr where it should contain a memberexpr). If i give it the type `array_ptr<HashEntry> array;` then i get lots of errors saying that `entry->array` has no bounds.